OutOfSchool/OutOfSchool.BackgroundJobs/Jobs/ElasticsearchSynchronizationQuartz.cs

```diff
@@ -20,11 +24,18 @@ public async Task Execute(IJobExecutionContext context)
         var elasticHealthService = scope.ServiceProvider.GetRequiredService<IElasticsearchHealthService>();
         if (elasticHealthService.IsHealthy)
         {
-            var elasticsearchSynchronizationService =
+            var config = scope.ServiceProvider.GetRequiredService<IOptions<ElasticConfig>>();
+
+            var elasticsearchWorkshopSynchronizationService =
+                scope.ServiceProvider
+                    .GetRequiredService<IElasticsearchSynchronizationService<IWorkshopService, Workshop>>();
+
+            var elasticsearchCompetitiveEventSynchronizationService =
                 scope.ServiceProvider
-                    .GetRequiredService<IElasticsearchSynchronizationService>();
+                    .GetRequiredService<IElasticsearchSynchronizationService<ICompetitiveEventService, CompetitiveEvent>>();
 
-            await elasticsearchSynchronizationService.Synchronize(context.CancellationToken).ConfigureAwait(false);
+            await elasticsearchWorkshopSynchronizationService.Synchronize(config.Value.WorkshopIndexName, context.CancellationToken).ConfigureAwait(false);
+            await elasticsearchCompetitiveEventSynchronizationService.Synchronize(config.Value.CompetitiveEventIndexName, context.CancellationToken).ConfigureAwait(false);
```

Comment:
I think it is better to have a separate job class for competitive events with different schedule, etc. It will allow us to have more control over when each job is executed.